### DISCUSSION TITLE
How to Duplicate MALT shader for multiple objects using script?

### DISCUSSION THREAD

**Discussion by krigeta** [bnpr/Malt#424] [Q&A]
Hello, I have a model where all the mesh parts are divided and I want to find a way via script so I can able to duplicate the MALT shader for every other mesh of the selected objects according to the mesh names.

here is a screenshot of what I am talking about.
![untitled](https://user-images.githubusercontent.com/75309361/196600522-831a42cb-c62f-430b-b000-20568fe6e155.png)

**Comment by Psi25** [comment]
You can just duplicate the material then plug in the Malt Node Tree. Duplicating the Malt Node Tree for different meshes is impractical.

**Reply by krigeta** [reply]
@Psi25 Okay, so, how can I just duplicate those materials only according to the meshes using script?

**Reply by Psi25** [reply]
You just duplicate the materials as if you're using Eevee or Cycles. However, materials in Malt works different since it's called shaders(node tree). Those shaders is what you put in the Materials.

**Reply by krigeta** [reply]
> You just duplicate the materials as if you're using Eevee or Cycles. However, materials in Malt work differently since it's called shaders(node tree). Those shaders are what you put in the Materials.

sir if you don't know the answer then, thanks.

**Reply by Psi25** [reply]
No problem.

**Comment by pragma37** [comment]
As @Psi25 mentioned, if you have multiple materials with the same nodes but different parameters (like color, etc.) it's best to use the same node tree and override those parameters in the material panel.

You could create a "reference" material with the node tree already assigned and then you can just duplicate it with the built-in copy function:
```
new_material = reference_material.copy()
new_material.name = "Your mesh name"
```

If you do it that way then there's nothing Malt specific you have to take into account when writing the script.
You could use the exact same code as you would for EEVEE/Cycles.
